feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radius_proxy.c

```diff
@@ -0,0 +1,231 @@
+		{
+			curl_easy_cleanup(curl);
+			fclose(fp);
+			return false;
+		}
+
+		curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0L);
+		curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0L);
+		curl_easy_setopt(curl, CURLOPT_HEADER, 0L);
+		curl_easy_setopt(curl, CURLOPT_URL, cert_url);
+		curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, file_write);
+		curl_easy_setopt(curl, CURLOPT_WRITEDATA, fp);
+		res = curl_easy_perform(curl);
+		curl_easy_cleanup(curl);
+		fclose(fp);
+		return res;
+	}
+
+	return true;
+}
+
+static bool raidus_proxy_config_set( struct schema_Radius_Proxy_Config *old,
+	struct schema_Radius_Proxy_Config *conf )
+{
+	int i=0;
+	char path[200];
+
+	blob_buf_init(&uci_buf, 0);
+	n = blobmsg_open_array(&uci_buf,"ListenUDP");
+	blobmsg_add_string(&uci_buf, NULL, "127.0.0.1:1812");
+	blobmsg_close_array(&uci_buf, n);
+	blob_to_uci_section(uci, "radsecproxy", NULL, "options", uci_buf.head, &radius_proxy_param, NULL);
+
+	blob_buf_init(&uci_buf, 0);
+	blobmsg_add_string(&uci_buf, "name", "localhost");
+	blobmsg_add_string(&uci_buf, "type", "udp");
+	blobmsg_add_string(&uci_buf, "secret", "secret");
+	blob_to_uci_section(uci, "radsecproxy", NULL, "client", uci_buf.head, &radius_proxy_param, NULL);
+
+	blob_buf_init(&uci_buf, 0);
+	if (strlen(conf->server))
+	{
+		blobmsg_add_string(&uci_buf, "name", conf->server);
+		blobmsg_add_string(&uci_buf, "type", "tls");
+		blobmsg_add_string(&uci_buf, "secret", "radsec");
+		blobmsg_add_bool(&uci_buf, "statusServer", 1);
+		blobmsg_add_bool(&uci_buf, "certificateNameCheck", 0);
+		blob_to_uci_section(uci, "radsecproxy", NULL, "server", uci_buf.head, &radius_proxy_param, NULL);
+	}
+
+	for (i = 0; i < conf->realm_len; i++)
+	{
+		blob_buf_init(&uci_buf, 0);
+		blobmsg_add_string(&uci_buf, "name", conf->realm[i]);
+		n = blobmsg_open_array(&uci_buf,"server");
+		blobmsg_add_string(&uci_buf, NULL, conf->server);
+		blobmsg_close_array(&uci_buf, n);
+		blob_to_uci_section(uci, "radsecproxy", NULL, "realm", uci_buf.head, &radius_proxy_param, NULL);
+	}
+
+	if (conf->radsec)
+	{
+		radsec_download_cert("cacert.pem", conf->radius_config_name, conf->ca_cert);
+		radsec_download_cert("clientcert.pem", conf->radius_config_name, conf->client_cert);
+		radsec_download_cert("clientdec.key", conf->radius_config_name, conf->client_key);
+
+		blob_buf_init(&uci_buf, 0);
+		blobmsg_add_string(&uci_buf, "name", conf->radius_config_name);
+		path[0] = '\0';
+		sprintf(path, "/usr/radsec/certs/%s/cacert.pem", conf->radius_config_name);
+		blobmsg_add_string(&uci_buf, "CACertificateFile", path);
+		path[0] = '\0';
+		sprintf(path, "/usr/radsec/certs/%s/clientcert.pem", conf->radius_config_name);
+		blobmsg_add_string(&uci_buf, "certificateFile", path);
+		path[0] = '\0';
+		sprintf(path, "/usr/radsec/certs/%s/clientdec.key", conf->radius_config_name);
+		blobmsg_add_string(&uci_buf, "certificateKeyFile", path);
+		blob_to_uci_section(uci, "radsecproxy", NULL, "tls", uci_buf.head, &radius_proxy_param, NULL);
+	}
+
```

Comment:
Do we need to do a uci_commit(), uci_unload() and reload_config =1 here before exiting?
Based on radius_proxy_config_delete() and callback_Radius_Proxy_Config() it seems required.

Reply:
Seems to work with uci_commit_all.